examples/inspection/plot_partial_dependence.py

```diff
@@ -163,179 +298,202 @@
+    "kind": "average",
+}
+_, ax = plt.subplots(ncols=3, figsize=(10, 4))
 tic = time()
-est_no_interactions = HistGradientBoostingRegressor(
-    interaction_cst=[[i] for i in range(X_train.shape[1])]
+display = PartialDependenceDisplay.from_estimator(
+    hgbdt_model,
+    X_train,
+    **features_info,
+    ax=ax,
+    **common_params,
 )
-est_no_interactions.fit(X_train, y_train)
 print(f"done in {time() - tic:.3f}s")
+_ = display.figure_.suptitle(
+    "1-way vs 2-way of numerical PDP using gradient boosting", fontsize=16
+)
+plt.subplots_adjust(wspace=0.3)
 
 # %%
-# The easiest way to show the effect of forbidden interactions is again the
-# ICE plots.
-
+# The two-way partial dependence plot shows the dependence of the number of bike rentals
+# on joint values of temperature and humidity.
+# We clearly see an interaction between the two features. For a temperature higher than
+# 20 degrees Celcius, the humidity will have a greater impact of the number of bike
+# rentals. For a temperature lower than 20 degrees Celsius, both the temperature and
+# humidity will have an impact on the number of bike rentals.
+#
+# We can reuse the gradient boosting model with interaction constraints between
+# features to see the difference:
 print("Computing partial dependence plots...")
+features_info = {
+    "features": ["temp", "humidity", ("temp", "humidity")],
+    "kind": "average",
+}
+_, ax = plt.subplots(ncols=3, figsize=(10, 4))
 tic = time()
 display = PartialDependenceDisplay.from_estimator(
-    est_no_interactions,
+    hgbdt_model_without_interactions,
     X_train,
-    ["MedInc", "AveOccup", "HouseAge", "AveRooms"],
-    kind="both",
-    subsample=50,
-    n_jobs=3,
-    grid_resolution=20,
-    random_state=0,
-    ice_lines_kw={"color": "tab:blue", "alpha": 0.2, "linewidth": 0.5},
-    pd_line_kw={"color": "tab:orange", "linestyle": "--"},
+    **features_info,
+    ax=ax,
+    **common_params,
 )
-
 print(f"done in {time() - tic:.3f}s")
-display.figure_.suptitle(
-    "Partial dependence of house value with Gradient Boosting\n"
-    "and no interactions allowed"
+_ = display.figure_.suptitle(
+    "1-way vs 2-way of numerical PDP using gradient boosting", fontsize=16
 )
-display.figure_.subplots_adjust(wspace=0.4, hspace=0.3)
+plt.subplots_adjust(wspace=0.3)
 
 # %%
-# All 4 plots have parallel ICE lines meaning there is no interaction in the
-# model.
-# Let us also have a look at the corresponding 2D-plot.
-
+# Although the 2D-plot shows much less interaction compared with the 2D-plot from
+# above, it is much harder to come to the conclusion that there is no interaction at
+# all. This might be a cause of the discrete predictions of trees in combination with
+# numerically precision of partial dependence. We also observe that the univariate
+# dependence plots have slightly changed as the model tries to compensate for the
+# forbidden interactions (spikes in the "humidity" PD plot).
```

Comment:
```suggestion
# The 1d partial dependence plots for the model constrained to not model feature
# interactions show local spikes for each features individually, in particular for
# for the "humidity" feature. Those spikes might be reflecting a degraded behavior
# of the model that attempts to somehow compensate for the forbidden interactions
# by overfitting particular training points. Note that the predictive performance
# of this model as measured on the test set is significantly worse than that of
# the original, unconstrained model.
#
# Also note that the number of local spikes visible on those plots is depends on
# the grid resolution parameter of the PD plot itself.
#
# Those local spikes result in a noisily gridded 2D PD plot. It is quite
# challenging to tell whether or not there are no interaction between those
# features because of the high frequency oscillations in the humidity feature.
# However it can clearly be seen that the simple interaction effect observed when
# the temperature crosses the 20 degrees boundary is no longer visible for this
# model.
```